includes/wikia/api/DesignSystemApiController.class.php

```diff
@@ -195,4 +203,57 @@ private function isWikiaOrgCommunity() {
 
 		return wfGetBaseDomainForHost( parse_url( $wgServer, PHP_URL_HOST ) ) === $wgWikiaOrgBaseDomain;
 	}
+
+
+	/**
+	 * External API request to IntentX
+	 */
+	public function getFandomShopDataFromIntentX() {
+		global $wgCityId, $wgMemc, $wgFandomShopMap, $wgFandomShopUrl;
+
+		$key = self::MEMC_PREFIX_FANDOM_STORE;
+		// get id from parameter or default to city id
+		$id = intval( $this->getVal( static::PARAM_ID, $wgCityId) );
+		// not using wfMemcKey here to avoid prefix conflicts
+		$memcKey = "intentX:$key:$id";
+
+		// don't make api call if not fandom store community
+		if ( !array_key_exists( $id, $wgFandomShopMap ) ) {
```

Comment:
Looks like the array keys are strings so the `$id` shouldn't be cast to an int like you did above. Just leave it as a string.